xmcl-keystone-ui/src/composables/modGroup.ts

```diff
@@ -274,8 +339,30 @@ export function useModGroups(isLocalView: Ref<boolean>, path: Ref<string>, items
     }
 
     instanceModGroupping.value = newVal
+    
+    // Ensure newly created groups from rules start collapsed
+    if (newlyCreatedGroups.length > 0) {
+      const newCollapsedState = { ...groupCollapsedState.value }
+      for (const groupName of newlyCreatedGroups) {
+        if (!(groupName in newCollapsedState)) {
+          newCollapsedState[groupName] = true // true means collapsed
+        }
+      }
+      groupCollapsedState.value = newCollapsedState
+    }
   }
 
+  // Auto-apply group rules when items change (e.g., when mods are enabled/disabled)
```

Comment:
Instead of reapply group rule to make enable/disable align, we should ensure when the item is disabled, the mod is still in the group. Which mean, when we try to determine the group of a file, it should ignore the .disabled suffix in fileName to lookup the group name.